### PR TITLE
Allow setting jenkins root url

### PR DESCRIPTION
**Pull request by chmouel**
Allow setting jenkins root URL directly from template,

### CODE REVIEW

**Comment by chmouel** [comment]
I haven't had the chance to fix up (gofabric8 deployment are too old currently) my local environment for testing this patch, so  let's hope to see if there is any issues

**Comment by aslakknutsen** [comment]
[test]

**Comment by fabric8cd** [comment]
@chmouel snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-529-1`

File: configuration/configuration.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ const (
 	varConsoleURL                      = "console.url"
 	varOpenshiftCheVersion             = "openshift.che.version"
 	varOpenshiftJenkinsVersion         = "openshift.jenkins.version"
+	varOpenshiftJenkinsRootURL         = "openshift.jenkins.root.url"
```
Comment (aslakknutsen) [inline configuration/configuration.go]: Please follow the naming from https://github.com/chmouel/fabric8-tenant/blob/0d6fadc590037b65564d7acddedb8daa728c5ce4/configuration/configuration.go#L45

File: configuration/configuration.go [hunk 2]
```diff
@@ -313,6 +314,11 @@ func (c *Data) GetOpenshiftJenkinsVersion() string {
 	return c.v.GetString(varOpenshiftJenkinsVersion)
 }
 
+// GetOpenshiftJenkinsRootURL returns the URL of the Jenkins URL
+func (c *Data) GetOpenshiftJenkinsRootURL() string {
```
Comment (aslakknutsen) [inline configuration/configuration.go]: No need for a Getter. Follow other 'pure' template vars: https://github.com/chmouel/fabric8-tenant/blob/0d6fadc590037b65564d7acddedb8daa728c5ce4/configuration/configuration.go#L366

File: configuration/configuration.go [hunk 3]
```diff
@@ -363,6 +369,7 @@ func (c *Data) GetTemplateValues() (map[string]string, error) {
 		"CHE_KEYCLOAK_AUTH__SERVER__URL": c.GetKeycloakURL() + "/auth",
 		"CHE_KEYCLOAK_REALM":             c.GetKeycloakRealm(),
 		"CHE_KEYCLOAK_CLIENT__ID":        c.GetKeycloakClientID(),
+		"JENKINS_ROOT_URL":               c.GetOpenshiftJenkinsRootURL(),
```
Comment (aslakknutsen) [inline configuration/configuration.go]: Need to add CM mapping in https://github.com/chmouel/fabric8-tenant/blob/master/OpenShiftTemplate.yml#L103 and some dummy data in the example: https://github.com/chmouel/fabric8-tenant/blob/master/f8tenant.secrets.yaml#L31

File: configuration/configuration.go (hunk 1 above)
Comment (chmouel) [reply configuration/configuration.go]: so something like varTemplateJenkinsRootURL instead? (not sure what is the recommender and if that's what you are pointing at)

**Comment by fabric8cd** [comment]
@chmouel snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-529-2`

**Comment by aslakknutsen** [comment]
[test]

**Pull request merged**